service/src/main/java/org/cbioportal/service/impl/StructuralVariantServiceImpl.java

```diff
@@ -131,4 +109,30 @@
         return countByGenes;
     }
 
+    @Override
+    public List<StructuralVariantCountByGene> getPatientCountInMultipleMolecularProfiles(
+            List<String> molecularProfileIds, List<String> patientIds, List<Integer> entrezGeneIds,
+            boolean includeFrequency, boolean includeMissingAlterationsFromGenePanel) {
+
+        List<StructuralVariantCountByGene> countByGenes = structuralVariantRepository
+                .getPatientCountInMultipleMolecularProfiles(molecularProfileIds, patientIds, entrezGeneIds);
+
+        // TODO: Remove once fusions are removed from mutation table
+        if (CollectionUtils.isEmpty(countByGenes)) {
```

Comment:
I'm glad this is getting removed, because this whole ID swap seems a bit janky. Can you explain why this is necessary, and what change in the future will obsolete this approach?

Reply:
This is related to https://github.com/cBioPortal/cbioportal/issues/6611. Currently all the fusion data is stored into mutation table with mutation molecular profile id. In future fusion data will be in its own table(structural variant) -> this is when all the import scripts are updated and existing data is migrated.

Reply:
I hope we haven't missed any TODO tags.